package NYT.NChunkClient.NProto;

import "yt_proto/yt/core/misc/proto/guid.proto";

////////////////////////////////////////////////////////////////////////////////

// TODO(achulkov2): [PForReview] Move these definitions into a separate medium descriptor proto file.
message TDomesticMediumDescriptor
{
}

message TS3MediumDescriptor
{
    optional bytes config = 1; // YSON
}

message TMediumDirectory
{
    message TMediumDescriptor
    {
        required string name = 1;
        required int32 index = 2;
        required int32 priority = 3;
        optional NYT.NProto.TGuid id = 4;

        oneof type_specific_descriptor {
            TDomesticMediumDescriptor domestic_medium_descriptor = 5;
            TS3MediumDescriptor s3_medium_descriptor = 6;
        }
    }

    repeated TMediumDescriptor medium_descriptors = 1;
}

////////////////////////////////////////////////////////////////////////////////
